server/events/plan_command_runner.go

```diff
@@ -241,7 +241,8 @@ func (p *PlanCommandRunner) run(ctx *command.Context, cmd *CommentCommand) {
 	if !cmd.IsForSpecificProject() {
 		ctx.Log.Debug("deleting previous plans and locks")
 		p.deletePlans(ctx)
-		_, err = p.lockingLocker.UnlockByPull(baseRepo.FullName, pull.Num)
+		// TODO(Ghais): make sure it's fine to remove this line:
+		//_, _, err = p.lockingLocker.UnlockByPull(baseRepo.FullName, pull.Num)
```

Comment:
@jamengual what do you think about removing this line? We wouldn't really benefit from removing all the locks here. As long as the plans are deleted with `p.deletePlans`, it should be fine to keep the locks assigned to the current pull requests.

Reply:
I think we need to bring @GenPage into this conversation but I do agree with you

Reply:
I'm not against it, but I have some concerns on the plan deletion code within pending_plan_finder, its a bit dated and it doesn't make sense to me while we handle deleting plans in multiple placces. Wouldn't it be better to use the deletePlan function in working_dir.go instead?